src/main/java/reactor/netty/tcp/TcpUtils.java

```diff
@@ -41,9 +41,7 @@ static Bootstrap updateProxySupport(Bootstrap b, ProxyProvider proxyOptions) {
 				NettyPipeline.ProxyHandler,
 				new ProxyProvider.DeferredProxySupport(proxyOptions));
 
-		if (b.config().resolver() == DefaultAddressResolverGroup.INSTANCE) {
-			return b.resolver(NoopAddressResolverGroup.INSTANCE);
-		}
+		BootstrapHandlers.updateResolverForProxySupport(b);
```

Comment:
This does not use the return value of the method. It's better to either mutate the passed in parameter OR return a value to be used, but not both IMHO.

Reply:
@anilgursel Yes just return the result from `BootstrapHandlers.updateResolverForProxySupport(b)`.
@ericbottard Just to clarify that we do not clone the `b`.

Reply:
The only reason I did this way is to follow the convention few lines above.  I am making the change now.